crates/sargon/src/wrapped_radix_engine_toolkit/low_level/manifest_summary/account_deposit/simple_fungible_resource_bounds.rs

```diff
@@ -0,0 +1,168 @@
+use crate::prelude::*;
+
+/// Represents the bounds for a simple fungible resource, which can
+/// be exact, at most, at least, between, or unknown amount.
+#[derive(Clone, Debug, PartialEq, Eq, uniffi::Enum)]
+pub enum SimpleFungibleResourceBounds {
+    Exact {
+        amount: Decimal,
+    },
+    AtMost {
+        amount: Decimal,
+    },
+    AtLeast {
+        amount: Decimal,
+    },
+    Between {
+        min_amount: Decimal,
+        max_amount: Decimal,
+    },
+    UnknownAmount,
+}
+
+impl SimpleFungibleResourceBounds {
+    pub fn exact(amount: Decimal) -> Self {
+        Self::Exact { amount }
+    }
+
+    pub fn at_most(amount: Decimal) -> Self {
+        Self::AtMost { amount }
+    }
+
+    pub fn at_least(amount: Decimal) -> Self {
+        Self::AtLeast { amount }
+    }
+
+    pub fn between(min_amount: Decimal, max_amount: Decimal) -> Self {
+        Self::Between {
```

Comment:
It would be invalid if `max < min` right? Please add such an assertion and document it - and add unit tests with `#[should_panic]`. 

Search for `# Panics` and you will see how Rust usually documents it :)